src/gui/editors/PianoRoll.cpp

```diff
@@ -797,14 +797,17 @@ void PianoRoll::constrainNoteLengths(bool constrainMax)
 	if (!hasValidMidiClip()) { return; }
 	m_midiClip->addJournalCheckPoint();
 
-	NoteVector notes = getSelectedNotes();
-	if (notes.empty())
+	const NoteVector selectedNotes = getSelectedNotes();
+
+	// TODO C++20: std::span
+	const NoteVector* notes = &selectedNotes;
+	if (selectedNotes.empty())
 	{
-		notes = m_midiClip->notes();
+		notes = &m_midiClip->notes();
 	}
```

Comment:
I think you can use a reference here with a ternary expression:
```cpp
const auto& notes = selectedNotes.empty() ? m_midiClip->notes() : selectedNotes;
```

Reply:
Don't know why I didn't think of that haha